### PR TITLE
TOOL-944 Check android codesign

### PR DESCRIPTION
**Pull request by lpusok**

### CODE REVIEW

File: phases/codesign.go [hunk 1]
```diff
@@ -83,43 +93,140 @@ func AutoCodesign(bitriseYML bitriseModels.BitriseDataModel, searchDir string) (
...
-				}}).run()
-
-			(&option{
-				title:  "Enter key password",
-				secret: true,
-				action: func(answer string) *option {
-					result.Android.KeyPassword = answer
-					return nil
-				}}).run()
+		absKeystorePath, err = pathutil.AbsPath(keystorePath)
```
Comment (trapacska) [inline phases/codesign.go]: no need to expand env vars and tilde here as bash does it already

File: phases/codesign.go [hunk 2]
```diff
@@ -83,43 +93,140 @@ func AutoCodesign(bitriseYML bitriseModels.BitriseDataModel, searchDir string) (
...
+		if err != nil {
+			return CodesignResultAndroid{}, fmt.Errorf("failed to get absolute keystore path, error: %s", err)
+		}
+		if _, err := os.Stat(absKeystorePath); os.IsNotExist(err) {
```
Comment (trapacska) [inline phases/codesign.go]: these checks should be a part of the validation checks

File: phases/codesign.go (hunk 1 above)
Comment (lpusok) [reply phases/codesign.go]: No in my experience it does not as we are not reading the path through bash.

File: phases/codesign.go (hunk 2 above)
Comment (lpusok) [reply phases/codesign.go]: If the path is wrong, I think it is better to fail early. The keystore validation would also fail, but only after reading in 3 additional paramters.

File: phases/codesign.go [hunk 3]
```diff
@@ -83,43 +93,140 @@ func AutoCodesign(bitriseYML bitriseModels.BitriseDataModel, searchDir string) (
...
+
+		"-alias",
+		codesign.Alias,
+		"-keypass",
+		codesign.KeyPassword,
+
+		"-J-Dfile.encoding=utf-8",
+		"-J-Duser.language=en-US",
+	}
+	cmd := command.New("keytool", params...)
```
[the remark this answers is not in the document]
Comment (lpusok) [reply phases/codesign.go]: What do you mean?

File: phases/codesign.go [hunk 4]
```diff
@@ -83,43 +93,140 @@ func AutoCodesign(bitriseYML bitriseModels.BitriseDataModel, searchDir string) (
...
+
+	out, err := cmd.RunAndReturnTrimmedCombinedOutput()
```
[the remark this answers is not in the document]
Comment (lpusok) [reply phases/codesign.go]: Currently the value of err is not printed if it only an exit status value, and the value of out is not printed if there is no out due to the command failed to run. So no unneeded/unspecified variables are printed.

File: phases/codesign.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (trapacska) [reply phases/codesign.go]: var, sorry

**Pull request merged**